### PR TITLE
Abort all cmd

### PR DESCRIPTION
**Pull request by andrewhare**
This adds the ability to abort all test runs in a single command.

### CODE REVIEW

File: cmd/testrun_abort.go [hunk 1]
```diff
@@ -48,3 +56,20 @@ func testRunAbort(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 }
+
+func testRunAbortAll(cmd *cobra.Command, args []string) {
+	c := NewClient()
+	ok, resp, err := c.TestRunAbortAll()
```
Comment (zeisss) [inline cmd/testrun_abort.go:62]: I guess we want to at least define the organisation that this command affects, right?

Comment (andrewhare) [reply cmd/testrun_abort.go:62]: Good point, at this point it aborts all runs for all orgs for the user.

File: api/testrun.go [hunk 2]
```diff
@@ -264,6 +264,31 @@ func (c *Client) TestRunAbort(testRunUID string) (bool, string, error) {
 	return response.StatusCode < 400, string(body), nil
 }
 
+// TestRunAbortAll sends a request to abort all running test cases.
+func (c *Client) TestRunAbortAll(organisationUID string) (bool, string, error) {
+	payload := url.Values{}
+	payload.Add("id", organisationUID)
```
Comment (zeisss) [inline api/testrun.go:270]:
```suggestion
	payload.Add("organisation_uid", organisationUID)
```

I think we should make it more clear what the `id` is. `TestCaseValidate()` sends an `organisation_uid`, so maybe we can use the same here.

File: cmd/testrun_abort.go [hunk 3]
```diff
@@ -21,10 +21,35 @@ var (
...
+		Short: "Abort all running tests for a given organisation",
+		Long:  "Abort all running tests for a given organisation",
+		Run:   testRunAbortAll,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if len(args) > 1 {
+				log.Fatal("Too many arguments")
+			}
+			if len(args) < 1 {
+				log.Fatal("Missing organisation")
+			}
```
Comment (zeisss) [inline cmd/testrun_abort.go:39]: You can make your life easier by settings the `Args` property ont he `cobra.Command` to `cobra.ExactArgs(1)`.

See https://pkg.go.dev/github.com/spf13/cobra#readme-positional-and-custom-arguments

File: cmd/testrun_abort.go [hunk 4]
```diff
@@ -48,3 +73,20 @@ func testRunAbort(cmd *cobra.Command, args []string) {
...
+	c := NewClient()
+	ok, resp, err := c.TestRunAbortAll(testRunAbortAllOpts.Organisation)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if ok {
+		fmt.Println(resp)
+		os.Exit(0)
+	}
```
Comment (zeisss) [inline cmd/testrun_abort.go:87]: Can we invert the case here so that the happy path is on the top level?

**Review by zeisss — CHANGES_REQUESTED** [review]

**Review by zeisss — APPROVED** [review]

**Pull request merged**